src/Voron/Impl/Paging/AbstractPager.cs

```diff
@@ -105,58 +94,91 @@ public void SetPagerState(PagerState newState)
             }
         }
 
+        protected void Lock(byte* address, long sizeToLock, TransactionState state)
+        {
+            if (PlatformDetails.RunningOnPosix == false)
+                // when running on linux we can't do anything from within the process, so let's avoid the locking entirely
+                WorkingSetIncreaseLocker.EnterUpgradeableReadLock();
```

Comment:
Why bother? There can only be a single one here anything, and there are no readers anyway

Reply:
`WorkingSetIncreaseLocker` is `static`.. the important thing here, is if we need to increase the WS, no other thread should try and lock the memory until the increasing thread is done.

Reply:
You are entering here with a single thread at a time, you should not use a reader / writer